pkgs/build-support/buildenv/builder.pl

```diff
@@ -111,7 +111,11 @@ sub findFiles {
 
     unless (-d $target && ($oldTarget eq "" || -d $oldTarget)) {
         if ($ignoreCollisions) {
-            warn "collision between `$target' and `$oldTarget'\n" if $ignoreCollisions == 1;
```

Comment:
What I also do not understand about the previous code is, why it checks first for `if ($ignoreCollisions)` and then has a second check `if $ignoreCollisions == 1;`. What other states can this variable has?

Reply:
This one stumps me, too.  I did a `git grep ignoreCollisions`, and every place it turned up was setting it to `true` or `false` -- in nix expressions.  So ... hopefully those translate to `1` and `0` in perl, but I'm not really sure.

Reply:
Found it:

```patch
commit 03f9026ac5ce7568c4b5c28d30552f869e229c63
Author: Eelco Dolstra <eelco.dolstra@logicblox.com>
Date:   Wed Jul 25 16:24:18 2012 -0400

    buildEnv: don't warn about collisions in propagated packages

    This mimicks buildenv in Nix more closely.

diff --git a/pkgs/build-support/buildenv/builder.pl b/pkgs/build-support/buildenv/builder.pl
index 71502c9f668..08331b178f4 100755
--- a/pkgs/build-support/buildenv/builder.pl
+++ b/pkgs/build-support/buildenv/builder.pl
@@ -66,7 +66,7 @@ sub findFiles {

     unless (-d $target && ($oldTarget eq "" || -d $oldTarget)) {
         if ($ignoreCollisions) {
-            warn "collision between `$target' and `$oldTarget'";
+            warn "collision between `$target' and `$oldTarget'" if $ignoreCollisions == 1;
             return;
         } else {
             die "collision between `$target' and `$oldTarget'";
@@ -122,7 +122,7 @@ while (scalar(keys %postponed) > 0) {
     my @pkgDirs = keys %postponed;
     %postponed = ();
     foreach my $pkgDir (sort @pkgDirs) {
-        addPkg($pkgDir, 1);
+        addPkg($pkgDir, 2);
     }
 }
```

The use of addPkg has changed since 2012, but there's still a case where `$ignoreCollisions` can be `2`. It's on line 198 currently.

Reply:
Therefore I think it makes sense to keep the `if $ignoreCollisions == 1` in both cases.

Reply:
It would be great to have a comment clarifying this.